traitsui/qt4/enum_editor.py

```diff
@@ -86,13 +92,14 @@ def init(self, parent):
             self._object, self._name, self._value = self.parse_extended_name(
                 factory.name
             )
-            self.values_changed()
+            self.values_changed(self._object)
             self._object.on_trait_change(
                 self._values_changed, " " + self._name, dispatch="ui"
             )
         else:
+            self.values_changed(factory)
```

Comment:
An alternative would be to define `self._value = lambda : self.factory.values` here, then the original `values_changed`  and `_values_changed` would continue to work. The added branches makes me think harder to ensure that the various combination of code paths are okay. In particular, `self._object` never raises `AttributeError` because this editor is a subclass of `HasPrivateTraits`, one gets `None` for any attribute access with the preceding "underscore" (another misfeature IMO).

Reply:
You're right, that is a much nicer approach. Thanks for the suggestion!